bundle/Controller/Admin/AdminController.php

```diff
@@ -271,10 +293,87 @@ public function handleCollectionAction(Request $request): RedirectResponse
         return $this->redirectToRoute('netgen_information_collection.route.admin.view', ['collectionId' => $collectionId]);
     }
 
+    /**
+     * @throws NonUniqueResultException
+     */
+    public function editAction(Request $request, int $collectionId): RedirectResponse|Response
+    {
+        $this->checkEditPermissions();
+
+        $collection = $this->service->getCollection(new CollectionId($collectionId));
+
+        $location = $collection->getContent()->contentInfo->getMainLocation();
+
+        $form = $this->builder->createUpdateFormForLocation($location, $collection)->getForm();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var InformationCollectionStruct $struct */
+            $struct = $form->getData()->payload;
+
+            /** @var ContentType $contentType */
+            $contentType = $form->getData()->definition;
+
+            $ezInfoCollection = $this->infoCollectionRepository->find($collectionId);
+
+            if ($ezInfoCollection === null) {
+                throw new NotFoundHttpException();
```

Comment:
Please use the shortcut method `createNotFoundException()`.

Reply:
fixed in commit d9710edd

Reply:
It still needs to be thrown :)

Reply:
fixed in 0c50e28d commit